client/context/query.go

```diff
@@ -82,22 +83,54 @@ func (ctx CLIContext) GetAccount(address []byte) (auth.Account, error) {
 }
 
 // GetFromAddress returns the from address from the context's name.
-func (ctx CLIContext) GetFromAddress() (from sdk.AccAddress, err error) {
-	if ctx.FromAddressName == "" {
-		return nil, errors.Errorf("must provide a from address name")
+func (ctx CLIContext) GetFromAddress() (sdk.AccAddress, error) {
+	if ctx.fromAddress == nil {
+		if err := ctx.populateFromFields(); err != nil {
+			return nil, err
+		}
+	}
+
+	return ctx.fromAddress, nil
+}
+
+// GetFromName returns the key name for the current context.
+func (ctx CLIContext) GetFromName() (string, error) {
+	if ctx.fromName == "" {
+		if err := ctx.populateFromFields(); err != nil {
+			return "", err
+		}
+	}
+
+	return ctx.fromName, nil
+}
+
+// NOTE: mutates state so must be pointer receiver
+func (ctx *CLIContext) populateFromFields() error {
```

Comment:
We'll need to think of another way implementing this functionality. The goal and point of contexts is that they are never modified, hence the `ctx = ctx.With*` logic. In our case here, maybe we'll need to to `ctx = populateFromFields(ctx)` somewhere upstream?

Reply:
Ok, I'll figure out a solution without the pointer receiver.